src/langcheck/augment/_common/_jailbreak_template.py

```diff
@@ -0,0 +1,60 @@
+from __future__ import annotations
+
+import random
+
+from langcheck.metrics.prompts._utils import get_template
+
+
+def _jailbreak_template(
+    instances: list[str] | str,
+    templates: list[str] | None,
+    available_templates: list[str],
+    language: str,
+    *,
+    num_perturbations: int = 1,
+) -> list[str]:
+    """Applies jailbreak templates to each string in instances.
+
+    Args:
+        instances: A single string or a list of strings to be augmented.
+        templates: A list of applied templates. If None, some templates are
+            randomly selected and used.
+        available_templates: A list of available templates.
+        language: The language of the templates.
+        num_perturbations: The number of perturbed instances to generate for
+            each string in instances. Should be equal to or less than the number
+            of templates.
+
+    Returns:
+        A list of perturbed instances.
+    """
+
+    instances = [instances] if isinstance(instances, str) else instances
+
+    if templates is None:
+        templates = available_templates
+
+    # Validation on the length of templates and num_perturbations
+    if num_perturbations > len(templates):
+        raise ValueError(
+            "The number of perturbations should be equal to or less than the number of templates."
+        )
+
+    # Validate that only available templates are specified
+    for template in templates:
+        if template not in available_templates:
+            raise ValueError(f"Invalid template: {template}")
+
+    perturbed_instances = []
+    for instance in instances:
+        # Randomly select num_perturbations templates
+        selected_templates = random.sample(templates, num_perturbations)
+        for template_name in selected_templates:
+            template = get_template(
+                f"{language}/jailbreak_templates/{template_name}.j2"
+            )
+            perturbed_instances.append(
+                template.render({"input_query": instance})
+            )
+
+    return perturbed_instances
```

Comment:
In addition to the `perturbed_instances`, maybe we can also return the list of corresponding templates? When testing this, I found myself wanting to know which template was used, and I can see that being pretty important if you want to calculate the success rate of different templates

Reply:
It is related to your suggestion on `num_perturbations`, but I tried to keep the interface consistent with the existence as much as possible. I thought about returning the template names as well, but it will offer very different interface as other functions in the module. In the worst case you can pass a template name one by one and concatenate them, so I prioritized the common interface compared to  the utility of the function itself.

WDYT? Do you still think it is important add these options?

Reply:
Yeah maybe returning an extra list isn't great since it'll be a different interface than the other `augment` functions. But I think allowing the randomization to be turned off (with `num_perturbations = None`) doesn't hurt?

Reply:
Wrote in the other comment, but to me it feels more natural to add another argument. Feel free to give me your opinion!

Reply:
(Not sure why the bottom thread is not being shown properly in Github, but fyi that I responded there!)
<img width="930" alt="image" src="https://github.com/user-attachments/assets/979ad404-e675-4e7b-a9ac-592e2f94dfe7">